src/google_oauth.erl

```diff
@@ -8,11 +8,15 @@
 
 get_access_token(ServiceAccountFile, Scope) when is_list(Scope) ->
     get_access_token(ServiceAccountFile, erlang:list_to_binary(Scope));
-get_access_token(ServiceAccountFile, Scope) ->
+get_access_token({file, ServiceAccountFile}, Scope) ->
     {ok, Bin} = file:read_file(ServiceAccountFile),
-    ServiceJson = jsx:decode(Bin, ?JSX_OPTS),
+    get_access_token({bin, Bin}, Scope);
+get_access_token({bin, ServiceAccountBin}, Scope) ->
+    ServiceJson = jsx:decode(ServiceAccountBin, ?JSX_OPTS),
```

Comment:
If I'm reading the code correctly you are double decoding here, first decode is in erlang-fcm:
https://github.com/2600hz/erlang-fcm/pull/2/files#diff-d4fc8cc3303765dc4105e6abeeb90d5c38ec42ba5bfaaf2e1c45de84b73a81b9R70

Reply:
@icehess the decode in `erlang-fcm` is only to pull out the `project_id` for use in that function. The `ServiceAccountBin` is, however, passed thru unmodified

Reply:
aah, in that case, you just pass the map and add a function header here to not requiring the decode it again. Just a thought, I know the object is small enough that it doesn't actually have any performance penalty.

Reply:
Added a `{map, ServiceJson}` option